fedimint-bitcoind/src/bitcoincore_rpc.rs

```diff
@@ -162,3 +182,81 @@ where
         })
     }
 }
+
+pub struct ElectrumClient(electrum_client::Client);
+
+impl ElectrumClient {
+    fn new(url: &Url) -> anyhow::Result<Self> {
+        Ok(Self(electrum_client::Client::new(url.as_str())?))
+    }
+}
+
+impl fmt::Debug for ElectrumClient {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.write_str("ElectrumClient")
+    }
+}
+use bitcoin_hashes::hex::ToHex;
+use electrum_client::ElectrumApi;
+
+#[async_trait]
+impl IBitcoindRpc for ElectrumClient {
+    async fn get_network(&self) -> Result<Network> {
+        let resp = fedimint_api::task::block_in_place(|| {
+            self.0.server_features().map_err(anyhow::Error::from)
+        })?;
+        Ok(match resp.genesis_hash.to_hex().as_str() {
+            "000000000019d6689c085ae165831e934ff763ae46a2a6c172b3f1b60a8ce26f" => Network::Bitcoin,
+            // https://blockstream.info/testnet/block/000000000933ea01ad0ee984209779baaec3ced90fa3f408719526f8d77f4943
+            "000000000933ea01ad0ee984209779baaec3ced90fa3f408719526f8d77f4943" => Network::Testnet,
+            // https://explorer.bc-2.jp/block/00000008819873e925422c1ff0f99f7cc9bbb232af63a077a480a3633bee1ef6
+            "00000008819873e925422c1ff0f99f7cc9bbb232af63a077a480a3633bee1ef6" => Network::Signet,
+            hash => {
+                warn!("Unknown genesis hash {hash} - assuming regtest");
+                Network::Regtest
+            }
+        })
+    }
+
+    async fn get_block_height(&self) -> Result<u64> {
+        fedimint_api::task::block_in_place(|| {
+            Ok(self.0.block_headers_subscribe_raw()?.height as u64)
+        })
+    }
+
+    async fn get_block_hash(&self, height: u64) -> Result<BlockHash> {
+        fedimint_api::task::block_in_place(|| {
+            Ok(self
+                .0
+                .block_headers(height as usize, 1)?
+                .headers
+                .get(0)
+                .ok_or_else(|| format_err!("empty block headers response"))?
+                .block_hash())
+        })
+    }
+
+    async fn get_block(&self, _hash: &BlockHash) -> Result<Block> {
+        unimplemented!()
+    }
+
+    async fn get_fee_rate(&self, confirmation_target: u16) -> Result<Option<Feerate>> {
+        fedimint_api::task::block_in_place(|| {
+            Ok(Some(Feerate {
+                sats_per_kvb: (self.0.estimate_fee(confirmation_target as usize)? * 100_000_000f64)
+                    .ceil() as u64,
+            }))
+        })
+    }
+
+    async fn submit_transaction(&self, transaction: Transaction) -> Result<()> {
+        fedimint_api::task::block_in_place(|| {
+            let mut bytes = vec![];
+            transaction
+                .consensus_encode(&mut bytes)
+                .expect("can't fail");
+            let _txid = self.0.transaction_broadcast_raw(&bytes)?;
+            Ok(())
+        })
+    }
```

Comment:
WTF! Where did the call go?!

Reply:
OK. Some of the tooling tricked me and stashed my change. I think I recovered everything, but be double suspicious of my code.